Modularization of metrics (#23936)
cilium-team/cilium: Documentation/observability/metrics.rst

+---------------------------------------------+-------------------------+--------+-----------+
|``api_limiter_wait_history_duration_seconds``|``api_call``             |Disabled|           |
+---------------------------------------------+-------------------------+--------+-----------+

API Limiter Labels
""""""""""""""""""

+-----------+--------------------------------------------------------------------+-----------+-----------------+
|Name       |Description                                                         |Known Value|Value Description|
+===========+====================================================================+===========+=================+
|``outcome``|Indicates whether the outcome of the operation was successful or not|``fail``   |                 |
|           |                                                                    +-----------+-----------------+
|           |                                                                    |``success``|                 |
+-----------+--------------------------------------------------------------------+-----------+-----------------+

eBPF
~~~~

> Both ``bpf_maps_virtual_memory_max_bytes`` and ``bpf_progs_virtual_memory_max_bytes`` are currently reporting the system-wide memory usage of eBPF that is directly and not directly managed by Cilium. This might change in the future and only report the eBPF memory usage directly managed by Cilium.

Reviewer: Avoid statements about the future. 
```suggestion
Both ``bpf_maps_virtual_memory_max_bytes`` and ``bpf_progs_virtual_memory_max_bytes`` are currently reporting the system-wide memory usage of eBPF that is directly and not directly managed by Cilium.
```

+--------------------------------+----------------------------------------+--------+-----------+
|``bpf_syscall_duration_seconds``|``operation``, ``outcome``              |Disabled|           |
+--------------------------------+----------------------------------------+--------+-----------+

eBPF Labels
"""""""""""

+-------------+--------------------------------------------------------------------+-----------+-----------------+
|Name         |Description                                                         |Known Value|Value Description|
+=============+====================================================================+===========+=================+
|``map_name`` |The label for the BPF map name                                      |           |                 |
+-------------+--------------------------------------------------------------------+-----------+-----------------+
|``operation``|The label for BPF maps operations                                   |           |                 |
+-------------+--------------------------------------------------------------------+-----------+-----------------+

> |``outcome``  |Indicates whether the outcome of the operation was successful or not|``fail``   |                 |

Reviewer: Use active voice
```suggestion
|``outcome``  |Indicates whether the operation succeeded or not|``fail``   |                 |
```

|Name         |Description                                                                                          |Known Value|Value Description|
+=============+=====================================================================================================+===========+=================+
|``area``     |Marks which area the metrics are related to (for example, which BPF map)                             |           |                 |
+-------------+-----------------------------------------------------------------------------------------------------+-----------+-----------------+
|``data``     |Marks the signal data                                                                                |           |                 |
+-------------+-----------------------------------------------------------------------------------------------------+-----------+-----------------+
|``direction``|The label for traffic direction                                                                      |           |                 |
+-------------+-----------------------------------------------------------------------------------------------------+-----------+-----------------+

> |``family``   |Marks which protocol family (IPv4, IPV6) the metric is related to.                                   |           |                 |
> +-------------+-----------------------------------------------------------------------------------------------------+-----------+-----------------+
> |``name``     |marks a unique identifier for this metric. The name should be defined once for a given type of error.|           |                 |
> +-------------+-----------------------------------------------------------------------------------------------------+-----------+-----------------+
> |``protocol`` |Marks the L4 protocol (TCP, ANY) for the metric.                                                     |           |                 |

Reviewer: Capitalize consistently. For fragments, use periods (or not) consistently
```suggestion
|``family``   |Marks which protocol family (IPv4, IPV6) the metric is related to                                   |           |                 |
+-------------+-----------------------------------------------------------------------------------------------------+-----------+-----------------+
|``name``     |Marks a unique identifier for this metric. The name should be defined once for a given type of error.|           |                 |
+-------------+-----------------------------------------------------------------------------------------------------+-----------+-----------------+
|``protocol`` |Marks the L4 protocol (TCP, ANY) for the metric                                                     |           |                 |
```

+-------------+-----------------------------------------------------------------------------------------------------+-----------+-----------------+
|``signal``   |Marks the signal name                                                                                |           |                 |
+-------------+-----------------------------------------------------------------------------------------------------+-----------+-----------------+

> |``status``   |The label from completed task                                                                        |           |                 |

Reviewer: Too fragmented
```suggestion
|``status``   |The label from the completed task                                                                        |           |                 |
```

+-----------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------+-----------------+
|``error``  |Indicates the type of error (string)                                                                                                                                                     |           |                 |
+-----------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------+-----------------+
|``kind``   |The kind of a label                                                                                                                                                                      |           |                 |
+-----------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------+-----------------+

> |``outcome``|Indicates whether the outcome of the operation was successful or not                                                                                                                     |``fail``   |                 |

Reviewer: Use active voice
```suggestion
|``outcome``|Indicates whether the operation succeeded or not                                                                                                                     |``fail``   |                 |
```

|``direction``       |The label for traffic direction                                                                                                                                                          |           |                                                   |
+--------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------+---------------------------------------------------+
|``enforcement``     |The label used to see the enforcement status                                                                                                                                             |           |                                                   |
+--------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------+---------------------------------------------------+

> |``family``          |Marks which protocol family (IPv4, IPV6) the metric is related to.                                                                                                                       |           |                                                   |

Reviewer: Use periods (or not) consistently
```suggestion
|``family``          |Marks which protocol family (IPv4, IPV6) the metric is related to                                                                                                                       |           |                                                   |
```

|``level``           |Log level                                                                                                                                                                                |``error``  |                                                   |
|                    |                                                                                                                                                                                         +-----------+---------------------------------------------------+
|                    |                                                                                                                                                                                         |``warning``|                                                   |
+--------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------+---------------------------------------------------+

> |``protocol``        |Marks the L4 protocol (TCP, ANY) for the metric.                                                                                                                                         |           |                                                   |
> +--------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------+---------------------------------------------------+
> |``protocol_l7``     |The label used when working with layer 7 protocols.                                                                                                                                      |           |                                                   |

Reviewer: Use periods (or not) consistently
```suggestion
|``protocol``        |Marks the L4 protocol (TCP, ANY) for the metric                                                                                                                                         |           |                                                   |
+--------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------+---------------------------------------------------+
|``protocol_l7``     |The label used when working with layer 7 protocols                                                                                                                                      |           |                                                   |
```

+--------------------------------+---------------+--------+-----------+
|``policy_l7_forwarded_total``   |               |Enabled |           |
+--------------------------------+---------------+--------+-----------+
|``policy_l7_parse_errors_total``|               |Enabled |           |
+--------------------------------+---------------+--------+-----------+
|``policy_l7_received_total``    |               |Enabled |           |
+--------------------------------+---------------+--------+-----------+
|``policy_l7_total``             |``rule``       |Enabled |           |
+--------------------------------+---------------+--------+-----------+
|``proxy_redirects``             |``protocol_l7``|Enabled |           |
+--------------------------------+---------------+--------+-----------+

Policy L7 (HTTP/Kafka) Labels
"""""""""""""""""""""""""""""

+---------------+---------------------------------------------------+-----------+-----------------+
|Name           |Description                                        |Known Value|Value Description|
+===============+===================================================+===========+=================+

> |``protocol_l7``|The label used when working with layer 7 protocols.|           |                 |

Reviewer: Use periods (or not) consistently
```suggestion
|``protocol_l7``|The label used when working with layer 7 protocols|           |                 |
```
